src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -195,6 +195,14 @@
             Ref("VariableAssignmentSegment"),
         ]
     ),
+    Expression_D_Potential_Select_Statement_Without_Brackets=ansi_dialect.get_grammar(
+        "Expression_D_Potential_Select_Statement_Without_Brackets"
+    ).copy(
+        insert=[
+            Ref("SessionVariableNameSegment"),
```

Comment:
I would like to add LocalVariableNameSegment, but I couldn't think of a good way to distinguish references to columns from references to variables. For the time being, we are only targeting SessionVariableNameSegment, which is easy to distinguish because it starts with @.